MarvelApiClient/build.gradle

```diff
@@ -14,22 +14,100 @@
  */
 
 apply plugin: 'java'
+apply plugin: 'maven'
+apply plugin: 'signing'
 
 sourceCompatibility = 1.7
 
 repositories {
-    mavenCentral()
+  mavenCentral()
 }
 
-dependencies {
-    testCompile group: 'junit', name: 'junit', version: '4.11'
-    testCompile 'org.mockito:mockito-all:1.9.5'
-    testCompile 'com.squareup.okhttp:mockwebserver:2.5.0'
-    testCompile 'commons-io:commons-io:2.4'
+def isReleaseBuild() {
+  return VERSION_NAME.contains("SNAPSHOT") == false
+}
+
+def getRepositoryUsername() {
+  return hasProperty('NEXUS_USERNAME') ? NEXUS_USERNAME : ""
+}
+
+def getRepositoryPassword() {
+  return hasProperty('NEXUS_PASSWORD') ? NEXUS_PASSWORD : ""
+}
+
+signing {
+  required { isReleaseBuild() && gradle.taskGraph.hasTask("uploadArchives") }
+  sign configurations.archives
+}
+
+task javadocJar(type: Jar) {
```

Comment:
I had the same feeling, but all people made it, I believe that you need both artifacts and this is the task for generate the java doc, if you have other idea or solution, I could change it, because my idea about gradle and maven is really basic.